src/Rover.cpp

```diff
@@ -94,32 +95,90 @@ void parseCommandLine(int argc, char** argv) {
 			  "LOG_WARN, LOG_ERROR)")
 		.default_value(std::string("LOG_INFO"))
 		.action([](const std::string& value) {
-			std::unordered_map<std::string, int> allowed{{"LOG_TRACE", LOG_TRACE},
-														 {"LOG_DEBUG", LOG_DEBUG},
-														 {"LOG_INFO", LOG_INFO},
-														 {"LOG_WARN", LOG_WARN},
-														 {"LOG_ERROR", LOG_ERROR}};
+			std::unordered_map<std::string, int> allowed{
+				{"LOG_ERROR", loguru::Verbosity_ERROR},
+				{"LOG_WARN", loguru::Verbosity_WARNING},
+				{"LOG_INFO", loguru::Verbosity_INFO},
+				{"LOG_DEBUG", 2},
+				{"LOG_TRACE", 1},
+			};
 
 			if (allowed.find(value) != allowed.end()) {
-				setLogLevel(allowed[value]);
-				return value;
+				loguru::g_stderr_verbosity = allowed[value];
 			}
 
 			throw std::runtime_error("Invalid log level " + value);
 		});
 
 	program.add_argument("-nc", "--no-colors")
 		.help("disables colors in console logging")
-		.action([&](const auto&) { setColorsEnabled(false); })
+		.action([&](const auto&) { loguru::g_colorlogtostderr = false; })
 		.nargs(0);
 
 	try {
+		loguru::init(argc, argv);
+		namespace fs = std::filesystem;
+
+		const auto LOG_LIFESPAN = std::chrono::hours(24 * 7);
+		try {
+			for (const auto& entry : fs::directory_iterator(fs::current_path())) {
+				std::cout << entry.path().extension();
+				if (entry.path().extension() == ".log" && entry.path().stem() != "latest") {
+					std::string dateString = entry.path().stem();
+
+					// Extract components from the date string
+					int year, month, day, hour, minute, second;
+					sscanf(dateString.c_str(), "%4d%2d%2d_%2d%2d%2d", &year, &month, &day,
+						   &hour, &minute, &second);
+
+					// Ensure scan outputs are correct
+					CHECK_EQ_F(std::stoi(dateString.substr(0, 4)), year);
+					CHECK_EQ_F(std::stoi(dateString.substr(4, 2)), month);
+					CHECK_EQ_F(std::stoi(dateString.substr(6, 2)), day);
+					CHECK_EQ_F(std::stoi(dateString.substr(9, 2)), hour);
+					CHECK_EQ_F(std::stoi(dateString.substr(11, 2)), minute);
+					CHECK_EQ_F(std::stoi(dateString.substr(13, 2)), second);
+
+					// Create a tm structure
+					std::tm tm_struct = {};
+					tm_struct.tm_year = year - 1900;
+					tm_struct.tm_mon = month - 1;
+					tm_struct.tm_mday = day;
+					tm_struct.tm_hour = hour;
+					tm_struct.tm_min = minute;
+					tm_struct.tm_sec = second;
+
+					// Current time
+					std::time_t unixTime = std::mktime(&tm_struct);
+
+					// Calculate duration
+					auto duration = std::chrono::system_clock::now() -
+									std::chrono::system_clock::from_time_t(unixTime);
+
+					// Delete log if it's older than LOG_LIFESPAN
+					if (duration > LOG_LIFESPAN) {
+						fs::remove(entry.path());
+					}
+				}
+			}
+		} catch (const fs::filesystem_error& e) {
+			std::cerr << "Error accessing current directory!" << e.what() << std::endl;
```

Comment:
Should probably add a space after the `!`. Might even want to do something like `std::cerr << "Error accessing current directory: " << e.what() << std::endl;`. Adding a space would be fine too. Just want it so that the string from `e.what()` is not just smashed up against the other string.